File: Calculator/android-native/app/src/main/java/com/qaantum/calculatorhub/calculators/LoanCalculator.kt
package com.qaantum.calculatorhub.calculators

import kotlin.math.pow

data class LoanResult(
    val monthlyPayment: Double,
    val totalInterest: Double,
    val totalCost: Double
)

class LoanCalculator {
    fun calculate(
        principal: Double,
        annualRate: Double,
        termMonths: Int
    ): LoanResult {
        val monthlyRate = annualRate / 100.0 / 12.0
        
        val monthlyPayment = if (monthlyRate == 0.0) {
            principal / termMonths
        } else {
            principal * (monthlyRate * (1 + monthlyRate).pow(termMonths)) /
                    ((1 + monthlyRate).pow(termMonths) - 1)
        }
        
        val totalPayment = monthlyPayment * termMonths
        val totalInterest = totalPayment - principal
        
        return LoanResult(
            monthlyPayment = monthlyPayment,
            totalInterest = totalInterest,
            totalCost = totalPayment
        )
    }
}

File: Calculator/ios-native/CalculatorHub/Calculators/LoanCalculator.swift
import Foundation

struct LoanResult {
    let monthlyPayment: Double
    let totalInterest: Double
    let totalCost: Double
}

class LoanCalculator {
    func calculate(principal: Double, annualRate: Double, termMonths: Int) -> LoanResult {
        let monthlyRate = annualRate / 100.0 / 12.0
        
        let monthlyPayment: Double
        if monthlyRate == 0 {
            monthlyPayment = principal / Double(termMonths)
        } else {
            monthlyPayment = principal * (monthlyRate * pow(1 + monthlyRate, Double(termMonths))) /
                (pow(1 + monthlyRate, Double(termMonths)) - 1)
        }
        
        let totalPayment = monthlyPayment * Double(termMonths)
        let totalInterest = totalPayment - principal
        
        return LoanResult(
            monthlyPayment: monthlyPayment,
            totalInterest: totalInterest,
            totalCost: totalPayment
        )
    }
}
